app/src/main/java/eu/kanade/tachiyomi/data/export/LibraryExporter.kt

```diff
@@ -0,0 +1,65 @@
+package eu.kanade.tachiyomi.data.export
+
+import android.content.Context
+import android.net.Uri
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import tachiyomi.domain.manga.model.Manga
+
+object LibraryExporter {
+
+    data class ExportOptions(
+        val includeTitle: Boolean,
+        val includeAuthor: Boolean,
+        val includeArtist: Boolean,
+    )
+
+    suspend fun exportToCsv(
+        context: Context,
+        uri: Uri,
+        favorites: List<Manga>,
+        options: ExportOptions,
+        onExportComplete: () -> Unit,
+    ) {
+        withContext(Dispatchers.IO) {
+            context.contentResolver.openOutputStream(uri)?.use { outputStream ->
+                val csvData = generateCsvData(favorites, options)
+                outputStream.write(csvData.toByteArray())
+                outputStream.flush()
+                onExportComplete()
+            }
+        }
+    }
+
+    private val escapeRequired = listOf("\r", "\n", "\"", ",")
+
+    private fun generateCsvData(favorites: List<Manga>, options: ExportOptions): String {
+        val columnSize = listOf(
+            options.includeTitle,
+            options.includeAuthor,
+            options.includeArtist,
+        )
+            .count { it }
+
+        val rows = buildList(favorites.size) {
+            favorites.forEach { manga ->
+                buildList(columnSize) {
+                    if (options.includeTitle) add(manga.title)
+                    if (options.includeAuthor) add(manga.author)
+                    if (options.includeArtist) add(manga.artist)
+                }
+                    .let(::add)
+            }
+        }
+        return rows.joinToString("\n") { columns ->
```

Comment:
https://datatracker.ietf.org/doc/html/rfc4180#section-2
```suggestion
        return rows.joinToString("\r\n") { columns ->
```